server/src/index.ts

```diff
@@ -3,13 +3,30 @@ import "reflect-metadata";
 import { Container } from "inversify";
 import { interfaces, InversifyRestifyServer, TYPE } from "inversify-restify-utils";
 import { BaseController } from "./controllers/BaseController";
+import { AuthenticationController } from "./controllers/AuthenticationController";
+import { plugins } from "restify";
+import { AuthenticationService } from "./services/AuthenticationService";
+import { AuthenticationRepository } from "./repositories/AuthenticationRepository";
 
 const container = new Container();
 container.bind<interfaces.Controller>(TYPE.Controller).to(BaseController).whenTargetNamed("BaseController");
+container
+    .bind<interfaces.Controller>(TYPE.Controller)
+    .to(AuthenticationController)
+    .whenTargetNamed("AuthenticationController");
+
+//register Repositories.
+
+container.bind(AuthenticationRepository).toSelf();
+
+//regsiter services.
+
+container.bind(AuthenticationService).toSelf();
```

Comment:
Services should be bound as a singleton

```suggestion
container
    .bind<AuthenticationService>('Service')
    .to(AuthenticationService)
    .inSingletonScope()
    .whenTargetNamed('AuthenticationService');
```